### PR TITLE
chore(obs): support the deleted check and refactor error check

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
The policy resource will throw error if the policy is not exist.
```
obs.ObsError{BaseModel:obs.BaseModel{StatusCode:404, RequestId:"00000189FD9A548944CD7D466841DD3F", ResponseHeaders:map[string][]string{"content-length":[]string{"327"}, "content-type":[]string{"application/xml"}, "date":[]string{"Wed, 16 Aug 2023 09:08:17 GMT"}, "id-2":[]string{"32AAAQAAEAABAAAQAAEAEAABCSLLxR4jwUOJ+yhGLYlqBQq5xkPTpqgV"}, "request-id":[]string{"00000189FD9A548944CD7D466841DD8F"}, "server":[]string{"OBS"}}}, Status:"404 Not Found", XMLName:xml.Name{Space:"", Local:"Error"}, Code:"NoSuchBucket", Message:"The specified bucket does not exist", Resource:"", HostId:"bF5gsx4Rft+zRXmejVqEWX14ARsH8awSQZ"}
```
The resource does not have deleted check logic.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support the deleted check.
2. refactor error check and setting logic.
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
NONE
```

捕获
捕获

### CODE REVIEW

File: huaweicloud/services/obs/resource_huaweicloud_obs_bucket_policy.go [hunk 1]
```diff
@@ -109,12 +111,14 @@ func resourceObsBucketPolicyRead(_ context.Context, d *schema.ResourceData, meta
 
 	log.Printf("[DEBUG] read policy for obs bucket: %s", d.Id())
 	output, err := obsClient.GetBucketPolicy(d.Id())
-
-	var pol string
-	if err == nil && output.Policy != "" {
-		pol = output.Policy
+	if err != nil {
+		if obsErr, ok := err.(obs.ObsError); ok && obsErr.BaseModel.StatusCode == 404 {
```
Comment (ShiChangkuo) [inline huaweicloud/services/obs/resource_huaweicloud_obs_bucket_policy.go]: use `obsErr.StatusCode` will be fine
```suggestion
		if obsErr, ok := err.(obs.ObsError); ok && obsErr.StatusCode == 404 {
```

File: huaweicloud/services/obs/resource_huaweicloud_obs_bucket.go [hunk 2]
```diff
@@ -576,17 +576,21 @@ func resourceObsBucketDelete(ctx context.Context, d *schema.ResourceData, meta i
...
 	if err != nil {
-		obsError, ok := err.(obs.ObsError)
-		if ok && obsError.Code == "BucketNotEmpty" {
-			log.Printf("[WARN] OBS bucket: %s is not empty", bucket)
-			if d.Get("force_destroy").(bool) {
-				err = deleteAllBucketObjects(obsClient, bucket)
-				if err == nil {
-					log.Printf("[WARN] all objects of %s have been deleted, and try again", bucket)
-					return resourceObsBucketDelete(ctx, d, meta)
+		if obsError, ok := err.(obs.ObsError); ok {
```
Comment (ShiChangkuo) [inline huaweicloud/services/obs/resource_huaweicloud_obs_bucket.go]: suggestion: you can write like this:
```go
    obsError, ok := err.(obs.ObsError)
    if !ok {
        return diag.Errorf("error deleting OBS bucket %s, %s", bucket, err)
    }

    if obsError.Code == "BucketNotEmpty" {
        ....
    }

    if obsError.StatusCode == 404 {
        ....
    }
```

File: huaweicloud/services/obs/resource_huaweicloud_obs_bucket_policy.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/obs/resource_huaweicloud_obs_bucket_policy.go]: all be done.

File: huaweicloud/services/obs/resource_huaweicloud_obs_bucket.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/obs/resource_huaweicloud_obs_bucket.go]: done.

**Comment by Lance52259** [comment]
delete obs bucket before running `terraform destroy`
删除前销毁
delete obs bucket before running `terraform refresh`
销毁并刷新
delete obs bucket when running `terraform destroy`
删除时销毁

**Pull request merged**